include/kernel/thread.h

```diff
@@ -234,7 +225,21 @@ static inline void updateTimestamp(void)
 {
     time_t prev = NODE_STATE(ksCurTime);
     NODE_STATE(ksCurTime) = getCurrentTime();
-    NODE_STATE(ksConsumed) += (NODE_STATE(ksCurTime) - prev);
+    time_t consumed = (NODE_STATE(ksCurTime) - prev);
+    NODE_STATE(ksConsumed) += consumed;
+    if (CONFIG_NUM_DOMAINS > 1) {
+
+        if ((consumed + MIN_BUDGET) >= ksDomainTime) {
```

Comment:
This forces a domain change once the domain time remaining would be < MIN_BUDGET.